builtin/providers/aws/resource_aws_launch_configuration.go

```diff
@@ -151,13 +164,20 @@ func resourceAwsLaunchConfigurationRead(d *schema.ResourceData, meta interface{}
 
 	lc := describConfs.LaunchConfigurations[0]
 
-	d.Set("key_name", lc.KeyName)
-	d.Set("iam_instance_profile", lc.IamInstanceProfile)
-	d.Set("image_id", lc.ImageId)
-	d.Set("instance_type", lc.InstanceType)
-	d.Set("name", lc.Name)
-	d.Set("security_groups", lc.SecurityGroups)
-	d.Set("spot_price", lc.SpotPrice)
+	d.Set("key_name", *lc.KeyName)
+	d.Set("image_id", *lc.ImageID)
+	d.Set("instance_type", *lc.InstanceType)
+	d.Set("name", *lc.LaunchConfigurationName)
+
+	if lc.IAMInstanceProfile != nil {
+		d.Set("iam_instance_profile", *lc.IAMInstanceProfile)
+	}
+	if lc.SpotPrice != nil {
+		d.Set("spot_price", *lc.SpotPrice)
+	}
+	if lc.SecurityGroups != nil {
+		d.Set("security_groups", lc.SecurityGroups)
+	}
```

Comment:
If these become unset during a read, then the state won't be updated. You might want to do a `else { zero value }` here. i.e.: `else { d.Set("iam_instance_profile", "") }`